dev/TabView/TabView.cpp

```diff
@@ -571,53 +604,77 @@ void TabView::UpdateTabWidths()
-                    auto const tabWidthForScroller = (availableWidth - (padding.Left + padding.Right)) / (double)(TabItems().Size());
+                    case winrt::TabViewWidthMode::Equal:
+                    {
+                        auto const minTabWidth = unbox_value<double>(SharedHelpers::FindInApplicationResources(c_tabViewItemMinWidthName, box_value(c_tabMinimumWidth)));
+                        auto const maxTabWidth = unbox_value<double>(SharedHelpers::FindInApplicationResources(c_tabViewItemMaxWidthName, box_value(c_tabMaximumWidth)));
 
-                    tabWidth = std::clamp(tabWidthForScroller, minTabWidth, maxTabWidth);
+                        // Calculate the proportional width of each tab given the width of the ScrollViewer.
+                        auto const padding = Padding();
+                        auto const tabWidthForScroller = (availableWidth - (padding.Left + padding.Right)) / (double)(TabItems().Size());
 
-                    // Size tab column to needed size
-                    tabColumn.MaxWidth(availableWidth);
-                    auto requiredWidth = tabWidth * TabItems().Size();
-                    if (requiredWidth >= availableWidth)
-                    {
-                        tabColumn.Width(winrt::GridLengthHelper::FromPixels(availableWidth));
-                        if (auto listview = m_listView.get())
+                        tabWidth = std::clamp(tabWidthForScroller, minTabWidth, maxTabWidth);
+
+                        // Size tab column to needed size
+                        tabColumn.MaxWidth(availableWidth);
+                        auto requiredWidth = tabWidth * TabItems().Size();
+                        if (requiredWidth >= availableWidth)
+                        {
+                            tabColumn.Width(winrt::GridLengthHelper::FromPixels(availableWidth));
+                            if (auto listview = m_listView.get())
+                            {
+                                winrt::FxScrollViewer::SetHorizontalScrollBarVisibility(listview, winrt::Windows::UI::Xaml::Controls::ScrollBarVisibility::Visible);
+                            }
+                        }
+                        else
                         {
-                            winrt::FxScrollViewer::SetHorizontalScrollBarVisibility(listview, winrt::Windows::UI::Xaml::Controls::ScrollBarVisibility::Visible);
+                            tabColumn.Width(winrt::GridLengthHelper::FromValueAndType(1.0, winrt::GridUnitType::Auto));
+                            if (auto listview = m_listView.get())
+                            {
+                                winrt::FxScrollViewer::SetHorizontalScrollBarVisibility(listview, winrt::Windows::UI::Xaml::Controls::ScrollBarVisibility::Hidden);
+                            }
                         }
                     }
-                    else
+                    break;
+
+                    case winrt::TabViewWidthMode::Compact:
```

Comment:
The Compact case is _almost_ the same as the SizeToContent case. Can they be combined? Or maybe even the same?

Reply:
Yes they behave the same, changed it into an if/else case instead of switch for that.